### DISCUSSION TITLE
Settings in VS 2022 for Pico4

### DISCUSSION THREAD

**Discussion by hnbeck** [StereoKit/StereoKit#994] [Q&A]
Hi, 
my platform is win11, VS2022 and Pico4. Although I use the working package for Android and mobile development, and the USB driver, VS does'nt recognize my Pico4 on USB.  Do I need additional settings, for example in the Android Device Manager Menu, or what could be the next steps to find the flaw?

Cheers

Hans

**Comment by maluoi** [comment]
First thing I'd recommend here is checking `adb devices` in command line, and make sure it shows up there! If it _doesn't_, then there's some environmental stuff you'll need to take care of, like making sure your Pico is in Developer Mode. If you can get the Pico to show up with `adb devices` then VS will also be able to see it :)

Developer Mode is the most likely culprit in my mind, but it never hurts to be aware that Visual Studio may start a _different_ `adb` executable than you're expecting, as VS has its own SDK. Doing an `adb kill-server` and then `adb devices` in the command line will kill VS's copy, and VS will continue using the one in your PATH instead. I haven't really seen issues related to this, but just in case!

There's also custom loader stuff, I don't remember the state of Pico 4, but if Pico hasn't moved to the official OpenXR Loader yet, you may also need to swap that out. You can read more about that over here: https://github.com/StereoKit/StereoKit/discussions/790 This is written towards the Snapdragon custom loader, but would be approximately the same for Pico!

**Comment by hnbeck** [comment]
Thanks for the  hints! 
adb devices gives 
`PA8150MGG9280022G       device`
as answer, which is the serial number of Pico
Pico is in developer mode, and I can see its file system in the Explorer. 
Android > The device log shows nothing, and 
Android>Device Manager shows a Google Pixel 3a with Android 14. (which is not there)
Is it possible that here is some flaw? My Pico runs on Android 10.....

Maybe it is very simple. In the stereokit docs it is mentioned "there is a option to deploy directly to the device". Where would I see this possibility in the VS UI, or where should my Pico be mentioned? I'm completly new to develop for Android.


Cheers

Hans

**Reply by maluoi** [reply]
Make sure you've got the Android project selected as your active project! By default, the template loads up with the Desktop project as active. Once the Android project is selected, you should see your device pop up right at the top like my Quest does here :)

![image](https://github.com/StereoKit/StereoKit/assets/8484884/703d20c6-d87c-4da7-be06-edca15105b66)

**Comment by rayperkins** [comment]
hey @hnbeck, I develop using a Pico 4 with Visual Studio and StereoKit. Here's a demo repo you can pull that should work as is, and it has the Pico OpenXR loader included: https://github.com/rayperkins/PicoSKDemo
Feel free to ping me if you need any help.

**Reply by hnbeck** [reply]
Thank you very much!! This is great :) Unfortunately, I cannot access the repo 
I'm looking forward to try it 

Best

Hans

**Reply by rayperkins** [reply]
Oops! Have made the repo public now.
